src/Tools/Microsoft.dotnet-openapi/src/Commands/BaseCommand.cs

```diff
@@ -0,0 +1,524 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+using Microsoft.DotNet.Openapi.Tools;
+using Microsoft.DotNet.Openapi.Tools.Internal;
+using Microsoft.Extensions.CommandLineUtils;
+
+namespace Microsoft.DotNet.OpenApi.Commands
+{
+    internal abstract class BaseCommand : CommandLineApplication
+    {
+        protected string WorkingDirectory;
+
+        protected readonly IHttpClientWrapper _httpClient;
+
+        public const string OpenApiReference = "OpenApiReference";
+        public const string OpenApiProjectReference = "OpenApiProjectReference";
+        protected const string SourceUrlAttrName = "SourceUrl";
+
+        public const string ContentDispositionHeaderName = "Content-Disposition";
+        private const string CodeGeneratorAttrName = "CodeGenerator";
+        private const string DefaultExtension = ".json";
+
+        internal const string PackageVersionUrl = "https://go.microsoft.com/fwlink/?linkid=2099561";
+
+        public BaseCommand(CommandLineApplication parent, string name, IHttpClientWrapper httpClient)
+        {
+            Parent = parent;
+            Name = name;
+            Out = parent.Out ?? Out;
+            Error = parent.Error ?? Error;
+            _httpClient = httpClient;
+
+            ProjectFileOption = Option("-p|--updateProject", "The project file update.", CommandOptionType.SingleValue);
+
+            Help = HelpOption("-?|-h|--help");
+            if (Parent is Application)
+            {
+                WorkingDirectory = ((Application)Parent).WorkingDirectory;
+            }
+            else
+            {
+                WorkingDirectory = ((Application)Parent.Parent).WorkingDirectory;
+            }
+
+            OnExecute(ExecuteAsync);
+        }
+
+        public CommandOption ProjectFileOption { get; }
+
+        internal CommandOption Help { get; }
+
+        public TextWriter Warning
+        {
+            get { return Out; }
+        }
+
+        protected abstract Task<int> ExecuteCoreAsync();
+
+        protected abstract bool ValidateArguments();
+
+        private async Task<int> ExecuteAsync()
+        {
+            if (!ValidateArguments() || Help.HasValue())
```

Comment:
Reverse this condition to avoid validating arguments when `--help` is specified.

Also, hasn't help already been shown at this point? May need to make the `OnExecute(...)` call in the constructor conditional on `!parent.IsShowingInformation` to avoid showing help twice.